ui/src/tests/pages/Login.test.tsx

```diff
@@ -0,0 +1,98 @@
+import {createRoot, Root} from 'react-dom/client';
+import {act} from 'react-dom/test-utils';
+import {MemoryRouter, Router} from 'react-router-dom';
+import {createMemoryHistory} from 'history';
+
+import {Login} from '../../pages/Login';
+
+let container: Element | null = null;
+let root: Root | null = null;
+
+jest.mock('../../globals/constants', () => ({
+  REDIRECT_URL: 'redirect-uri',
+  GOOGLE_CLIENT_ID: 'google-client-id',
+}));
+
+jest.mock('../../assets', () => ({
+  logo: 'dummy-logo',
+  loginArrow: 'dummy-arrow',
+}));
+
+describe('<Login />', () => {
+  const {location} = window;
+
+  beforeAll(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    delete (window as any).location;
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    window.location = {assign: jest.fn()} as any;
+  });
+
+  beforeEach(() => {
+    act(() => {
+      root = createRoot(container!);
+    });
+    window.sessionStorage.clear();
+  });
+
+  it('should redirect to google oauth2 page on login click', () => {
+    act(() => {
+      root!.render(
+        <MemoryRouter>
+          <Login />
+        </MemoryRouter>
+      );
+    });
+
+    const loginButton = document.querySelector('[data-testid=signin]');
+    expect(loginButton!.textContent!.trim()).toBe('SIGN IN');
```

Comment:
Do we need to check this here? This test should only check if the `sign in` button redirects to the google OAuth consent screen.

Reply:
You're right. Added a separate test for checking if the component renders properly.